### PR TITLE
runfix: Use child observables properly for user details and avatar

### PR DESCRIPTION
**Pull request by AndyLnd**

### CODE REVIEW

File: src/script/util/ComponentUtil.ts [hunk 1]
```diff
@@ -69,3 +69,30 @@ export const useKoSubscribable = <T = any>(observable: ko.Subscribable<T>, defau
...
+  const getInitialState = (root: ParentOfObservables): ChildValues =>
+    children.reduce((acc, child) => {
+      acc[child] = root[child]();
+      return acc;
+    }, {} as ChildValues);
+
+  const [state, setState] = useState<ChildValues>(getInitialState(parent));
+  useEffect(() => {
+    setState(getInitialState(parent));
+    const subscriptions = [] as ko.Subscription[];
```
Comment (ffflorian) [inline src/script/util/ComponentUtil.ts]:
```suggestion
    const subscriptions: ko.Subscription[] = [];
```

File: src/script/util/ComponentUtil.ts [hunk 2]
```diff
@@ -69,3 +69,30 @@ export const useKoSubscribable = <T = any>(observable: ko.Subscribable<T>, defau
...
+    children.forEach(child => {
+      subscriptions.push(
```
Comment (ffflorian) [inline src/script/util/ComponentUtil.ts]: Why not use a `map` here?

**Review by ffflorian — CHANGES_REQUESTED** [review]

File: src/script/util/ComponentUtil.ts (hunk 2 above)
Comment (AndyLnd) [reply src/script/util/ComponentUtil.ts]: 🤦🏻‍♂️ Of course. I added the map later and didn't think of it at that point.

Comment (ffflorian) [reply src/script/util/ComponentUtil.ts]: :)

**Review by thisisamir98 — APPROVED** [review]

**Review by ffflorian — APPROVED** [review]

**Pull request merged**